### ISSUE TITLE
Incorrect fatal error when compact encoding is ineffective

### ISSUE THREAD

**Issue by Daniel-Cortez** [pawn-lang/compiler#425]
### Issue description:
The compiler issues an incorrect error for the following code:
```Pawn
main()
{
    // NOTE: In compact encoding '0xAABBCCDD' can't be compressed well
    // and takes 5 bytes instead of 4, thus making the compression ineffective.
    static const a[32] = { 0xAABBCCDD, ... };
    return a[0];
}
```
Since the compression shouldn't be effective for this code, one would expect the compiler printing the following message:
```
warning 232: output file is written, but with compact encoding disabled
```
But instead it issues a fatal error (and aborts compilation as a result):
```
fatal error 106: compiled script exceeds the maximum memory size (1521262808 bytes)
```
And instead of `1521262808` there might be any other (seemingly random) value.

### Minimal complete verifiable example (MCVE):
See above.

### Workspace Information:

* Compiler version: 3.2.3664, 3.10.9
* Command line arguments provided (or sampctl version):
* Operating System:

**Comment by Daniel-Cortez** [comment]
I tried debugging and it seems that the error comes from the following code in function `assemble()` (`sc6.c`):
```C
  if (bytes_out-bytes_in>0)
    error(106);         /* compression buffer overflow */
```
Which means the error is triggered when
* the compact encoding is ineffective (`bytes_out > bytes_in`), and
* the output size is bigger than the input, but only by less than 62 bytes (`bytes_out-bytes_in < AMX_COMPACTMARGIN-2`) (if the difference is 62 or more bytes, the error would rather be in `write_encoded()` or `write_encoded_n()`).

And that code looks suspicious for the following reasons:
* The format text for error 106 is `compiled script exceeds the maximum memory size (%ld bytes)`, which implies that function `error()` should take at least two arguments for this error, but in the abovementioned code it only takes one.
* The comment at that line says `compression buffer overflow`, which doesn't correspond with the actual error text (see above).
* And guess what? Previously error 106 indeed was related to compression - in Small compiler (at least in Small 2.7.3 it was phrased as `compaction buffer overflow`).

So all of this means that at some point (probably when Small was renamed to Pawn) ID 106 was reassigned to a different error (instead in Pawn compiler there's warning 232: `output file is written, but with compact encoding disabled`), but the abovementioned code wasn't updated in a corresponding way.
I think the correct code should be:
```C
  if (bytes_out-bytes_in>0)
    longjmp(compact_err,1);
```

**Comment by Daniel-Cortez** [comment]
Fixed in #423.
